vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -308,14 +323,22 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
           let firstNeededPage = page;
           let lastNeededPage = page;
           for (let idx = firstNeededIndex; idx <= lastNeededIndex; idx++) {
-            firstNeededPage = Math.min(
-              firstNeededPage,
-              grid.$connector._getPageIfSameLevel(parentKey, idx, firstNeededPage)
-            );
-            lastNeededPage = Math.max(
-              lastNeededPage,
-              grid.$connector._getPageIfSameLevel(parentKey, idx, lastNeededPage)
+            const potentialSameLevelCacheAndIndex = grid._cache.getCacheAndIndex(idx);
+            const potentialSameLevelItem =
+              potentialSameLevelCacheAndIndex.cache.items[potentialSameLevelCacheAndIndex.scaledIndex];
+            if (!potentialSameLevelItem) {
```

Comment:
I believe the index should be included in the range regardless of whether the associated item is already loaded or not. The item can be still loading when another page on a parent level is requested.